lib/src/play_store_search_api.dart

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2021 William Kwabla. All rights reserved.
+ */
+
+import 'package:html/dom.dart';
+import 'package:html/parser.dart' show parse;
+import 'package:http/http.dart' as http;
+
+class PlayStoreSearchAPI {
+  /// Play Store Search Api URL
+  final String playStorePrefixURL = 'play.google.com';
+
+  /// Provide an HTTP Client that can be replaced for mock testing.
+  http.Client? client = http.Client();
+
+  bool debugEnabled = false;
+
+  /// Look up by id.
+  Future<Document?> lookupById(String id) async {
+    if (id.isEmpty) {
+      return null;
+    }
+
+    final url = lookupURLById(id)!;
+
+    final response = await http.get(Uri.parse(url));
+
+    if (response.statusCode != 200) {
+      print('Can\'t find an app in the Play Store with the id: $id');
+      return null;
+    }
+
+    final decodedResults = _decodeResults(response.body);
+
+    return decodedResults;
+  }
+
+  String? lookupURLById(id) {
+    final url =
+        Uri.https(playStorePrefixURL, '/store/apps/details', {'id': '$id'})
+            .toString();
+
+    return url;
+  }
+
+  Document? _decodeResults(jsonResponse) {
+    if (jsonResponse.isNotEmpty) {
+      final decodedResults = parse(jsonResponse);
+
+      return decodedResults;
+    }
+    return null;
+  }
+}
+
+class PlayStoreResults {
+  /// Return field releaseNotes from Play Store results.
+  static String? releaseNotes(response) {
+    final sectionElements = response.getElementsByClassName('W4P4ne');
+    final releaseNotesElement = sectionElements.firstWhere(
+      (elm) => elm.querySelector('.wSaTQd')!.text == 'What\'s New',
+    );
+    final releaseNotes = releaseNotesElement
+        .querySelector('.PHBdkd')
+        ?.querySelector('.DWPxHb')
+        ?.text;
+
+    return releaseNotes;
+  }
+
+  /// Return field trackViewUrl from Play Store results.
+  static String? trackViewUrl(id) {
+    final uri =
+        Uri.https('play.google.com', '/store/apps/details', {'id': '$id'})
+            .toString();
+
+    return uri;
+  }
+
+  /// Return field version from Play Store results.
+  static String? version(response) {
+    final additionalInfoElements = response.getElementsByClassName('hAyfc');
+    final versionElement = additionalInfoElements.firstWhere(
+      (elm) => elm.querySelector('.BgcNfc')!.text == 'Current Version',
+    );
+
+    final storeVersion = versionElement.querySelector('.htlgb')!.text;
```

Comment:
@Williano This does not work for app `com.google.android.apps.maps`. It returns `Varies with device`.

Reply:
Most of the google apps and some apps doesn’t show current version on play store because Google Play allows you to publish different APKs for your application. Each one targeted to different device configuration. Thus, each APK is an independent version of your application, but they share the same application listing on Google Play and must share the same package name and be signed with the same release key.

Usually Android applications run on most compatible devices with a single APK, by supplying alternative resources for different configurations (e.g., different layouts for different screen sizes) and the Android system selects the appropriate resources for the device at runtime. But in a few cases, however, a single APK is unable to support all device configurations, because alternative resources make the APK file too big (greater than 50MB) or other technical challenges prevent a single APK from working on all devices.

**And even though Google doesn't encourage developers to use this "multiple APK" feature, most Google apps use it: Google+, Google Chrome, Gmail, Google Maps, Google Search, Google Play Books, Google Play Movies & TV, Google Translate**